### ISSUE TITLE
Seems ecowitt integration is broken

### ISSUE THREAD

**Issue by edegraaff** [garbled1/homeassistant_ecowitt#60]
Since my latest upgrade of homeassistant ecowitt integration fails to load and start.

Ecowitt Weather Station integration for home-assistant viA hacs

Systeemstatus
Versie	core-2021.6.2
Type installatie	Home Assistant Container
Ontwikkeling	false
Supervisor	false
Docker	true
Virtuele omgeving	false
Python-versie	3.8.9
Besturingssysteem	Linux
Versie van het besturingssysteem	4.4.59+
CPU-architectuur	x86_64
Tijdzone	Europe/Amsterdam

In the HACs reops the ecowitt integration is there, but when i try to add it in homeassistant after reinstall, the integration component is not shown or availible in ha,

it seems the installation is not registered somewheren.

it worked without any problems form months now

**Comment by kkellner** [comment]
Likely you are encountering this issue:  https://github.com/garbled1/homeassistant_ecowitt/issues/43

Manually adding the version to the `manifest.json` file worked for me.

**Comment by edegraaff** [comment]
Thanks, ok, in that case, i am running in HA in a docker, i think the fastest way is to jump back one version and/or wait for an update. I tried to change the file in the bash terminal but the file is read only.

**Comment by Janthenman67** [comment]
edegraaf, it's really easy to solve, a bit sub-optimal  to go back a version and wait for the update if you ask me, I work with containers as wel, here is what to do:.
just open file editor >> custom_components>> ecowiitt
just edit in the manifest.jason: 
"version": "7.5.1",
and restart HA
done, worked for me

**Comment by applehomekit1** [comment]
Not working here for me either, and I have tried adding the version number to no avail. Any ideas?
Using core-2021.6.3 on an intel NUC. All other HACS installs are working fine.

**Comment by Janthenman67** [comment]
could you copy/paste your maifest.jason file?

**Comment by DaDre** [comment]
I have the same issue. I removed the integration (both in HACS and Configurations) while trying to diagnose the issue. I have since re-added it to HACS successfully, but can't add it to integrations in configurations as its not coming up in search at all. Rebooted and tried various things many times.

Manifest file below:

`{
  "domain": "ecowitt",
  "name": "Ecowitt Weather Station",
  "version": "1.2.3",
  "config_flow": true,
  "documentation": "https://www.home-assistant.io/integrations/ecowitt",
  "requirements": [
    "pyecowitt==0.11"
  ],
  "ssdp": [],
  "zeroconf": [],
  "homekit": {},
  "dependencies": [],
  "codeowners": [
    "@garbled1"
  ]
}`

**Comment by Janthenman67** [comment]
2 spaces before "pyecowitt==0.11"
also before: "@garbled1"
and ]

![image](https://user-images.githubusercontent.com/69641492/121210090-adf55080-c87b-11eb-9c41-d3e3980e0337.png)

**Comment by DaDre** [comment]
Updated my previous post with a screenshot of manifest file for clarity. Edited file to match 7.5.1 (just in case). Still same issue. Integration does not show up when searching for it.

**Comment by RedWheiler80** [comment]
Go to issue #49 

Probably there is your solution described.

**Comment by DaDre** [comment]
> Go to issue #59
> 
> Probably there is your solution described.
> 
> Basically like MarcelD1973 stated:
> 
> Restart HomeAssistant
> **Manually delete** all the files and maps in Custom Components / ecowitt
> Restart HA
> Re-install ecowitt from HACS
> Fill in the “version”
> Restart HA
> 
> And my (working in 2021.6.3) example of the manifest.json file:
> 
> ```
> 
> {
>   "domain": "ecowitt",
>   "name": "Ecowitt Weather Station",
>   "version": "7.5.1",
>   "config_flow": true,
>   "documentation": "https://www.home-assistant.io/integrations/ecowitt",
>   "requirements": [
>     "pyecowitt==0.11"
>   ],
>   "ssdp": [],
>   "zeroconf": [],
>   "homekit": {},
>   "dependencies": [],
>   "codeowners": [
>     "***@***.***"
>   ]
> }
> ```

I tried this. Still no go.

**Comment by flofricke** [comment]
@edegraaff Try to delete the browser cache. I had exactly the same problem a few days ago. Repo in HACS but no integration component. After deleting it showed up.

**Comment by davidjbwatson** [comment]
cannot get ecowitt integration . last core upgrade no ecowitt, tried suggestions none work

**Comment by applehomekit1** [comment]
I updated to core-2021.6.3 and when it restarted, the integration was working. Not sure what else was I had done, but all is good for now.

**Comment by jlbphotos** [comment]
I had same issue, did all the steps no luck, updated to 2021.6.4 and now working fine. (still modified my version)

**Comment by fun4stuff** [comment]
I am having the same problems. Sorry for the stupid question, but when you say you updated the core to 6.4, is that for homeassistant or for Ecowitt firmware? How do i do this?

**Comment by Janthenman67** [comment]
Are you using the docker image from Frederike?

**Comment by fun4stuff** [comment]
I am using docker image: homeassistant/raspberrypi3-homeassistant:stable

But i got it to work! I stopped the docker container, removed docker container, pulled latest homeassistant docker container (2021.6.5), restarted HA docker container then repeated steps above: delete everything in Ecowitt directory, restart HA, reinstall Ecowitt in HACS, add version in json file, and restart HA. Then I was able to add Ecowitt integration and all is working. 

Thanks!
